torchmultimodal/modules/encoders/text_encoder.py

```diff
@@ -0,0 +1,155 @@
+        token_type_ids (Tensor, optional): Tensor of input token type ids of shape [batch, seq_len]. In BERT,
+            used to indicate whether a word is in sentence A or B for next sentence prediction
+        position_ids (Tensor, optional): Tensor of input position ids of shape [batch, seq_len]
+        inputs_embeds (Tensor, optional): Tensor of input embeddings of shape [batch, hidden_size],
+            if embeddings are calculated elsewhere
+
+    Raises:
+        ValueError: if input_ids and inputs_embeds are both None
+    """
+
+    def __init__(
+        self,
+        embeddings: nn.Module,
+        encoder: nn.Module,
+        layernorm: Optional[nn.Module] = None,
+        pooler: Optional[nn.Module] = None,
+        weight_init_fn: Optional[Callable] = None,
+    ) -> None:
+        super().__init__()
+
+        self.embeddings = embeddings
+        self.encoder = encoder
+        self.layernorm = layernorm
+        self.pooler = pooler
+
+        if weight_init_fn:
+            self.apply(weight_init_fn)
+
+    def forward(
+        self,
+        input_ids: Optional[Tensor] = None,
+        attention_mask: Optional[Tensor] = None,
+        token_type_ids: Optional[Tensor] = None,
+        position_ids: Optional[Tensor] = None,
+        inputs_embeds: Optional[Tensor] = None,
+    ) -> TransformerOutput:
+        if input_ids is not None:
+            input_shape = input_ids.size()
+            device = input_ids.device
+        elif inputs_embeds is not None:
+            input_shape = inputs_embeds.size()[:-1]
+            device = inputs_embeds.device
+        else:
+            raise ValueError("input_ids or inputs_embeds must not be None")
+
+        # only mask out padding token if no mask specified
+        if attention_mask is None:
+            attention_mask = torch.ones(input_shape, device=device)
+            attention_mask[input_ids == self.embeddings.pad_token_id] = 0
+
+        # massage attention mask to correct shape for transformer
+        attention_mask = get_extended_attention_mask(attention_mask)
+
+        embedding_output = self.embeddings(
+            input_ids=input_ids,
+            position_ids=position_ids,
+            token_type_ids=token_type_ids,
+            inputs_embeds=inputs_embeds,
+        )
+        encoder_output = self.encoder(embedding_output, attention_mask=attention_mask)
+
+        sequence_output = encoder_output.last_hidden_state
+        pooled_output = encoder_output.pooler_output
+        if self.layernorm:
+            sequence_output = self.layernorm(sequence_output)
+        if self.pooler:
+            pooled_output = self.pooler(sequence_output)
+
+        return TransformerOutput(
+            last_hidden_state=sequence_output,
+            pooler_output=pooled_output,
+            hidden_states=encoder_output.hidden_states,
+            attentions=encoder_output.attentions,
+        )
+
+
+def text_encoder(
```

Comment:
Add a comment explaining what the default values correspond to. I assume BERT, but which one?